lazy-pages/src/utils.rs

```diff
@@ -41,44 +41,41 @@ pub(crate) fn with_inclusive_ranges<P: PageNumber, E>(
     // `pages` is a BTreeSet, which is ordered, so all panics are unreachable.
 
     for &page in pages_iter {
-        let after_end = end
-            .raw()
+        let after_end = PageNumber::raw(&end)
             .checked_add(1)
             .unwrap_or_else(|| unreachable!("`end` must be smaller than page, so inc must be ok"));
-        if after_end != page.raw() {
-            let iter = start
-                .iter_end_inclusive(end)
+        if after_end != PageNumber::raw(&page) {
+            let iter = PageNumber::iter_end_inclusive(&start, end)
                 .unwrap_or_else(|| unreachable!("`end` must be bigger or equal to start"));
             f(iter)?;
             start = page;
         }
         end = page;
     }
 
-    let iter = start
-        .iter_end_inclusive(end)
+    let iter = PageNumber::iter_end_inclusive(&start, end)
         .unwrap_or_else(|| unreachable!("`end` must be bigger or equal than `start`"));
 
     f(iter)
 }
 
 #[cfg(test)]
 mod tests {
-    use crate::pages::{GearPageNumber, PageDynSize, PageNumber, PagesIterInclusive};
+    use gear_core::pages::{GearPage, PageDynSize, PageNumber, PagesIterInclusive};
 
     #[test]
     fn test_with_inclusive_range() {
         let test = |pages: &[u32]| {
             let mut inclusive_ranges: Vec<Vec<u32>> = Vec::new();
-            let slice_to_ranges = |iter: PagesIterInclusive<GearPageNumber>| -> Result<(), ()> {
+            let slice_to_ranges = |iter: PagesIterInclusive<GearPage>| -> Result<(), ()> {
                 inclusive_ranges.push(iter.map(|p| p.raw()).collect());
                 Ok(())
             };
 
             super::with_inclusive_ranges(
                 &pages
                     .iter()
-                    .map(|p| GearPageNumber::new(*p, &10).unwrap())
+                    .map(|p| GearPage::new(*p, &10).unwrap())
```

Comment:
I think better to use here now test for PageU32Size. This will look more naturally))